logger.h

```diff
@@ -17,22 +17,84 @@
 #define TO_S(M) #M
 #define STR(M) TO_S(M)
 
+#ifndef _LOG_NO_STD
+#ifdef _LOG_STDOUT
+#define LOG_STD(LEVEL, f_str, ...) fprintf(stdout, "[" STR(LEVEL) "]["  __FILE__  ":" STR(__LINE__) "] " f_str, ##__VA_ARGS__)
+#endif // _LOG_STDOUT
+#ifndef _LOG_STDOUT
+#define LOG_STD(LEVEL, f_str, ...) fprintf(stderr, "[" STR(LEVEL) "]["  __FILE__  ":" STR(__LINE__) "] " f_str, ##__VA_ARGS__)
+#endif // _LOG_STDOUT
+#else
+#define LOG_STD(LEVEL, f_str, ...)
+#endif // _LOG_NO_STD
+
+#ifdef _SYS_LOG
+#define LOG_SYS(LEVEL, f_str, ...) syslog(3, "[" STR(LEVEL) "]["  __FILE__  ":" STR(__LINE__) "] " f_str, ##__VA_ARGS__)
+#else
+#define LOG_SYS(LEVEL, f_str, ...)
+#endif // _SYS_LOG
+
+#ifdef _LOG_RUNTIME_ENVIRONMENT
+extern FILE* user_file;
+#define LOG_UMSG(LEVEL, f_str, ...) fprintf(user_file, "[" STR(LEVEL) "]["  __FILE__  ":" STR(__LINE__)
```

Comment:
The string that is printed to the log is same for all loggers, please create some abstraction that will define it only once.